libcaf_net/caf/net/transport_worker_dispatcher.hpp

```diff
@@ -65,48 +68,48 @@ class transport_worker_dispatcher {
 
   template <class Parent>
   error handle_data(Parent& parent, span<byte> data, id_type id) {
-    auto it = workers_by_id_.find(id);
-    if (it == workers_by_id_.end()) {
-      // TODO: where to get node_id from here?
+    auto worker = find_by_id(id);
+    if (!worker) {
+      // TODO: where to get id_type from here?
       add_new_worker(parent, node_id{}, id);
-      it = workers_by_id_.find(id);
+      worker = find_by_id(id);
     }
-    auto worker = it->second;
     return worker->handle_data(parent, data);
   }
 
   template <class Parent>
   void write_message(Parent& parent,
                      std::unique_ptr<endpoint_manager_queue::message> msg) {
-    auto sender = msg->msg->sender;
-    if (!sender)
+    auto receiver = msg->receiver;
+    if (!receiver)
       return;
-    auto nid = sender->node();
-    auto it = workers_by_node_.find(nid);
-    if (it == workers_by_node_.end()) {
+    auto nid = receiver->node();
+    auto worker = find_by_node(nid);
+    if (!worker) {
       // TODO: where to get id_type from here?
       add_new_worker(parent, nid, id_type{});
-      it = workers_by_node_.find(nid);
+      worker = find_by_node(nid);
     }
```

Comment:
Stay DRY: this duplicates code from `handle_data` and should thus go into a function.

Reply:
The problem is, that I need to distinguish between `node_id` and `id_type`. I refactored the code now, but I don't see how not to duplicate this code. Could you have a look at the changed parts?